lib/src/binary_gray.dart

```diff
@@ -0,0 +1,152 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// binary_gray.dart
+// A converter that can transform binary values into Gray code values or Gray
+// code values into binary values.
+//
+// 2023 October 5
+// Author: Yao Jing Quek <yao.jing.quek@intel.com>
+
+import 'package:rohd/rohd.dart';
+
+/// A module for converting binary values to Gray code representation.
+///
+/// The [BinaryToGrayConverter] class represents a module that takes a binary
+/// value as input and produces the equivalent Gray code representation as
+/// output. It internally uses the [binaryToGrayMap] function for the
+/// conversion.
+class BinaryToGrayConverter extends Module {
+  /// The Gray code representation output of the converter.
+  ///
+  /// This [Logic] value represents the Gray code representation of the binary
+  /// input provided to the converter. It can be accessed using
+  /// the [grayCode] getter.
+  Logic get grayCode => output('gray_code');
+
+  /// Creates a [BinaryToGrayConverter] instance with the specified binary
+  /// input.
+  ///
+  /// The [binary] parameter is the binary input that you want to convert
+  /// to Gray code. The width of the input [binary] determines the width
+  /// of the Gray code output.
+  BinaryToGrayConverter(Logic binary) {
+    final inputWidth = binary.width;
+    binary = addInput('binary_input', binary, width: inputWidth);
+    final grayVal = addOutput('gray_code', width: inputWidth);
+
+    Combinational([
+      Case(
+          binary,
+          [
+            for (var i = 0; i < (1 << inputWidth); i++)
+              CaseItem(Const(i, width: inputWidth), [
+                grayVal <
+                    Const(
+                        binaryToGrayMap(LogicValue.ofInt(i, inputWidth))
+                            .swizzle(),
+                        width: inputWidth)
+              ])
+          ],
+          conditionalType: ConditionalType.unique)
+    ]);
+  }
+
+  /// Converts a binary value represented by [binary] to Gray code.
+  ///
+  /// Given a [binary] value, this function returns a list of [LogicValue]
+  /// representing the equivalent Gray code.
+  ///
+  /// For each bit in the [binary] input, starting from the least significant
+  /// bit (index 0), the function calculates the corresponding Gray code bit
+  /// based on XOR operation with the previous bit. The resulting Gray code
+  /// bits are returned in a list.
+  ///
+  /// If [binary] has no bits, an empty list is returned.
+  ///
+  /// Returns a list of [LogicValue] representing the Gray code.
+  List<LogicValue> binaryToGrayMap(LogicValue binary) {
```

Comment:
could this be made into a `static` function?  And we can call it just `binaryToGray`?  Seems like a useful function to expose.